src/NuGet.Clients/NuGet.Console/OutputConsole/BuildOutputConsole.cs

```diff
@@ -51,10 +52,14 @@ public void Clear()
         {
         }
 
-        [SuppressMessage("Microsoft.VisualStudio.Threading.Analyzers", "VSTHRD010", Justification = "NuGet/Home#4833 Baseline")]
         public void Write(string text)
         {
-            _outputWindowPane?.OutputStringThreadSafe(text);
+            NuGetUIThreadHelper.JoinableTaskFactory.Run(async delegate
+            {
+                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                _outputWindowPane?.OutputStringThreadSafe(text);
```

Comment:
can we check _outputWindowPane for null before doing JTF.Run?

Reply:
sure, but its just a fail-safe otherwise it's never null there